test/integration_test/migration_dr_actions_test.go

```diff
@@ -0,0 +1,548 @@
+	sourceDeploymentReplicas := *sourceDeployments.Items[0].Spec.Replicas
+
+	// Create the clusterpair
+	clusterPairNamespace := mysqlNamespace
+	if !unidirectionalClusterpair {
+		log.Info("Bidirectional flag is set, will create bidirectional cluster pair:")
+		log.InfoD("Name: %s", remotePairName)
+		log.InfoD("Namespace: %s", clusterPairNamespace)
+		log.InfoD("Backuplocation: %s", defaultBackupLocation)
+		log.InfoD("Secret name: %s", defaultSecretName)
+		err = scheduleBidirectionalClusterPair(remotePairName, clusterPairNamespace, projectIDMappings, defaultBackupLocation, defaultSecretName)
+		log.FailOnError(t, err, "failed to set bidirectional cluster pair: %v", err)
+		err = setSourceKubeConfig()
+		log.FailOnError(t, err, "failed to set kubeconfig to source cluster: %v", err)
+	} else {
+		log.Info("Creating Unidirectional flag is set, will create unidirectional cluster pair:")
+		log.InfoD("Name: %s", remotePairName)
+		log.InfoD("Namespace: %s", clusterPairNamespace)
+		log.InfoD("Backuplocation: %s", defaultBackupLocation)
+		log.InfoD("Secret name: %s", defaultSecretName)
+		err = scheduleUnidirectionalClusterPair(remotePairName, clusterPairNamespace, projectIDMappings, defaultBackupLocation, defaultSecretName, true, false)
+		log.FailOnError(t, err, "failed to set unidirectional cluster pair: %v", err)
+	}
+
+	// Create the migration schedule
+	// we want to create the schedulePolicy and migrationSchedule using storkctl instead of scheduling apps using torpedo's scheduler
+	// schedulePolicyArgs is a map of schedulePolicyName : {{flag1:value1,flag2:value2,....}}
+	schedulePolicyArgs := make(map[string]map[string]string)
+	schedulePolicyArgs["migrate-every-5m"] = map[string]string{"policy-type": "Interval", "interval-minutes": "5"}
+
+	// migrationScheduleArgs is a map of migrationScheduleName : {{flag1:value1,flag2:value2,....}}
+	migrationScheduleArgs := make(map[string]map[string]string)
+	migrationScheduleArgs[instanceID] = map[string]string{
+		"purge-deleted-resources": "",
+		"schedule-policy-name":    "migrate-every-5m",
+	}
+
+	//Create schedulePolicies using storkCtl if any required
+	factory := storkctl.NewFactory()
+	var outputBuffer bytes.Buffer
+	cmd := storkctl.NewCommand(factory, os.Stdin, &outputBuffer, os.Stderr)
+	for schedulePolicyName, customArgs := range schedulePolicyArgs {
+		cmdArgs := []string{"create", "schedulepolicy", schedulePolicyName}
+		executeStorkCtlCommand(t, cmd, cmdArgs, customArgs)
+	}
+	//Create migrationSchedules using storkCtl
+	migrationScheduleName := "forward-migration-schedule-running-migration"
+	cmdArgs := []string{"create", "migrationschedule", migrationScheduleName, "-c", remotePairName,
+		"--namespaces", mysqlNamespace, "-n", mysqlNamespace}
+	executeStorkCtlCommand(t, cmd, cmdArgs, migrationScheduleArgs[migrationKey])
+
+	// bump time of the world by 1 minutes
+	mockNow := time.Now().Add(1 * time.Minute)
```

Comment:
one doubt..
how does this ensure migration is still running ??
why should we have a time gap?

Reply:
In this test , ValidateMigrationSchedule is not called and there is a 1 minute sleep before creating failover action, that way we are ensuring either a Pending/Running always happens. The idea here is not to have a completed migration which is unlikley in 1 minute.